src/SourceLocator/Ast/Strategy/NodeToReflection.php

```diff
@@ -36,6 +36,8 @@ public function __invoke(
         }
 
         if ($node instanceof Node\FunctionLike) {
+            /** @var Node\Stmt\ClassMethod|Node\Stmt\Function_|Node\Expr\Closure $node */
+            $node = $node;
```

Comment:
Same: why is this being fone with an assignment? Also, should the instanceof check verify these 3 types?

Reply:
```
        if ($node instanceof Node\Stmt\ClassMethod
            || $node instanceof Node\Stmt\Function_
            || $node instanceof Node\Expr\Closure
        ) {
```

You're right, this is definitely better.